libs/core/src/lib/vertical-navigation/vertical-navigation.component.ts

```diff
@@ -26,38 +28,72 @@ import { VerticalNavigationMainNavigationComponent } from './vertical-navigation
 export class VerticalNavigationComponent implements AfterContentInit, OnDestroy {
     /** Whether or not this component is to be shown in 'condensed' mode. */
     @Input()
-    condensed = false;
-
-    /** An RxJS Subject that will kill the data stream upon component’s destruction (for unsubscribing)  */
-    private readonly _onDestroy$ = new Subject<void>();
+    set condensed(value: boolean) {
+        this._condensed.set(value);
+    }
 
-    /** An RxJS Subject that will kill the data stream upon queryList changes (for unsubscribing)  */
-    private readonly _onRefresh$ = new Subject<void>();
+    get condensed(): boolean {
+        return this._condensed();
+    }
 
     /** @hidden */
     @ContentChildren(ListNavigationItemComponent, { descendants: true })
-    private _navigationItems: QueryList<ListNavigationItemComponent>;
+    private readonly _navigationItems: QueryList<ListNavigationItemComponent>;
 
     /** @hidden
      * Querylist of list-items in main navigation.
      */
     @ContentChildren(ListNavigationItemComponent)
-    private _mainNavigationItems: QueryList<ListNavigationItemComponent>;
+    private set _mainItems(items: QueryList<ListNavigationItemComponent>) {
+        this._mainNavItems.set(items?.toArray() ?? []);
+    }
 
     /** @hidden */
     private _keyManager: FocusKeyManager<ListNavigationItemComponent>;
 
+    /** An RxJS Subject that will kill the data stream upon component’s destruction (for unsubscribing)  */
+    private readonly _onDestroy$ = new Subject<void>();
+
+    /** An RxJS Subject that will kill the data stream upon queryList changes (for unsubscribing)  */
+    private readonly _onRefresh$ = new Subject<void>();
+
+    /** @hidden */
+    private readonly _condensed = signal(false);
+
+    /** @hidden */
+    private readonly _contentInited = signal(false);
+
+    /** @hidden */
+    private readonly _mainNavItems = signal<ListNavigationItemComponent[]>([]);
+
+    /** @hidden */
+    constructor() {
+        effect(
+            () => {
+                if (!this._contentInited()) {
+                    return;
+                }
+
+                const isCondensed = this._condensed();
+
+                this._mainNavItems().forEach((item) => {
+                    item._condensed.set(isCondensed);
+                });
+            },
+            {
+                allowSignalWrites: true
+            }
+        );
```

Comment:
Can we make a injection from mainNavItem to navbar and have calculation from there? I just do not like signal writes in effects

Reply:
It should be done on the root level only, so we use contentCHildren with descendants false in this case. As to writting signals in effects, it still ok to do so